File: 1305-all-elements-in-two-binary-search-trees/1305-all-elements-in-two-binary-search-trees.py
class Solution(object):
    def getAllElements(self, root1, root2):
        def trverse(root):
            ans, dq = [], deque([root])
            while dq:
                node = dq.pop()
                if type(node) == int: ans.append(node)
                elif node: dq.extend([node.left, node.val, node.right])
            return ans

        l1, l2, ans = trverse(root1), trverse(root2), []
        while l1 and l2: ans.append(l1.pop() if l1[-1] < l2[-1] else l2.pop())
        return ans + list(reversed(l1 or l2))

File: 1305-all-elements-in-two-binary-search-trees/1305-all-elements-in-two-binary-search-trees.cpp
class Solution {
public:
    void dfs(TreeNode* root, vector<int>& ans){
        if (root==NULL)return;
        if (root->left)dfs(root->left,ans);
        if (root->right)dfs(root->right,ans);
        ans.push_back(root->val);
    }
    vector<int> getAllElements(TreeNode* root1, TreeNode* root2) {
        vector<int> ans;
        dfs(root1,ans);
        dfs(root2,ans);
        sort(ans.begin(),ans.end());
        return ans;
    }
};
